approval_polls/static/approval_polls/create.js

```diff
@@ -18,4 +18,35 @@ $(function () {
 
   $('[data-toggle=tooltip]').tooltip();
   $('button#add-choice').click($.proxy(this.addChoiceField, this));
+
+  function roundMinutes(date) {
+    if (date.getMinutes() > 0 && date.getMinutes() < 30){
+      date.setHours(date.getHours());
+      date.setMinutes(30);
+    }
+    else if (date.getMinutes() > 30 && date.getMinutes() < 60){
+      date.setHours(date.getHours()+1);
+      date.setMinutes(0);
+    }
+    return date;
+  }
+
+  var minDateTimelogic = function( currentDateTime ){
+    this.setOptions({
+      minDate: roundMinutes(currentDateTime),
+      minTime: currentDateTime,
+    });
+  };
+
+  $('#datetimepicker').datetimepicker({
+    step:30,
+    onShow:minDateTimelogic,
+  });
+
+  $("#datetimepicker").keyup(function (e)
+  {
+    if(e.keyCode == 8 || e.keyCode == 46) {
+        $(this).val("");
+    }
+  });
```

Comment:
This is a nice idea to have the datetimepicker completely cleared. Though I am able to type random characters into the datetimepicker, although the control will not create a poll with an invalid date as you have taken care of it. Perhaps you could also prevent all other characters from being typed into the datetimepicker with something like: 
```
    $('#datetimepicker').keypress(function(e) {
      e.preventDefault();
    });
```